docs/src/literate/topology_optimization.jl

```diff
@@ -0,0 +1,502 @@
+        χn[i] = state.χ
+    end
+    return pΨ, χn 
+end
+#md nothing # hide
+
+# Now we calculate the Laplacian. For this purpose, we will later create the grid topology of 
+# the grid by using the function `ExclusiveTopology`. Then we iterate through each face of each element,
+# obtaining the neighboring element by using the `getneighborhood` function. For boundary faces,
+# the function call will return an empty object. In that case we use the dictionary to instead find the opposite
+# face, as discussed in the introduction. Then, the approximation of the Laplacian reduces to the sum below. 
+
+function Laplacian(dh, topology, χn, Δh)
+    ∇²χ = zeros(getncells(dh.grid))
+    _nfaces = nfaces(dh.grid.cells[1])
+    opp = Dict(1=>3, 2=>4, 3=>1, 4=>2)
+    nbg = zeros(Int,_nfaces)
+    
+    for element in CellIterator(dh)
+        i = cellid(element)
+        for j in 1:_nfaces
+            nbg_cellid = getcells(getneighborhood(topology, dh.grid, FaceIndex(i,j)))
+            if(!isempty(nbg_cellid))
+                nbg[j] = first(nbg_cellid) ## assuming only one face neighbor per cell
+            else ## boundary face
+                nbg[j] = first(getcells(getneighborhood(topology, dh.grid, FaceIndex(i,opp[j]))))
+            end
+        end
+        
+        ∇²χ[i] = (χn[nbg[1]]+χn[nbg[2]]+χn[nbg[3]]+χn[nbg[4]]-4*χn[i])/(Δh^2)
+    end
+
+    return ∇²χ
+end
+#md nothing # hide
+
+# To ensure that the mass is kept constant, we have to calculate the constraint
+# parameter $\lambda$, which we do via the bisection method. We repeat the calculation
+# until the difference between the current density and target density nearly vanishes. 
+
+function Bisection(χn, Δχ, ρ, ηs, χ_min) 
+    n_el = length(χn)
+    
+    χ_trial = zeros(n_el)
+    ρ_trial = 0.0
+    
+    λ_lower = minimum(Δχ) - ηs
+    λ_upper = maximum(Δχ) + ηs
+    λ_trial = 0.0
+    
+    while(abs(ρ-ρ_trial)>1e-7)
+        for i in 1:n_el
+            Δχt = 1/ηs * (Δχ[i] - λ_trial)
+            χ_trial[i] = maximum([χ_min, minimum([1.0, χn[i]+Δχt])])
+        end
+    
+        ρ_trial = 0.0
+        for i in 1:n_el
+            ρ_trial += χ_trial[i]/n_el 
+        end
+        
+        if(ρ_trial > ρ)
+            λ_lower = λ_trial
+        elseif(ρ_trial < ρ)
+            λ_upper = λ_trial
+        end
+        λ_trial = 1/2*(λ_upper+λ_lower)
+    end
+
+    return χ_trial
+end
+#md nothing # hide
+
+# Finally, we put everything together to update the density. We weight our numerical parameters 
+# $\eta$ and $\beta$ by the average driving forces to make the choice of the parameters independent
+# from the specific problem. 
+
+function UpdateDensity(dh, states, mp, ρ, topology, Δh)   
+    pΨ, χn = DrivingForces(states, mp, dh) ## driving forces, old density field
+    ∇²χ = Laplacian(dh, topology, χn, Δh) ## Laplacian
+
+    p_Ω = sum(pΨ)/length(pΨ) ## average driving force
+
+    Δχ = pΨ + mp.β*p_Ω*∇²χ
+    
+    χn1 = Bisection(χn, Δχ, ρ, mp.η*p_Ω, mp.χ_min)
```

Comment:
```suggestion
    χn1 = compute_χn1(χn, Δχ, ρ, mp.η*p_Ω, mp.χ_min)
```

Reply:
See above, call the function compute_bisection(..)?